packages/cli/src/Server.ts

```diff
@@ -1693,197 +1708,200 @@ class App {
 
-					// We sadly get nothing back from "update". Neither if it updated a record
-					// nor the new value. So query now the hopefully updated entry.
-					const responseData = await Db.collections.Credentials!.findOne(id);
+				// @ts-ignore
+				credentials.setData(updateData.data, encryptionKey);
 
-					if (responseData === undefined) {
-						throw new ResponseHelper.ResponseError(
-							`Credentials with id "${id}" could not be found to be updated.`,
-							undefined,
-							400,
-						);
-					}
+				const newCredentialData = credentials.getDataToSave() as ICredentialsDb;
 
-					// Remove the encrypted data as it is not needed in the frontend
-					responseData.data = '';
+				// Add special database related data
+				newCredentialData.updatedAt = this.getCurrentDate();
 
-					// Convert to response format in which the id is a string
-					(responseData as unknown as ICredentialsResponse).id = responseData.id.toString();
-					return responseData as unknown as ICredentialsResponse;
-				},
-			),
+				await this.externalHooks.run('credentials.update', [newCredentialData]);
+
+				// Update the credentials in DB
+				await Db.collections.Credentials!.update(credentialId, newCredentialData);
+
+				// We sadly get nothing back from "update". Neither if it updated a record
+				// nor the new value. So query now the hopefully updated entry.
+				const responseData = await Db.collections.Credentials!.findOne(credentialId);
+
+				if (responseData === undefined) {
+					throw new ResponseHelper.ResponseError(
+						`Credential with ID "${credentialId}" could not be found to be updated.`,
+						undefined,
+						400,
+					);
+				}
+
+				// Remove the encrypted data as it is not needed in the frontend
+				const { id, data, ...rest } = responseData;
+
+				return {
+					id: id.toString(),
+					...rest,
+				};
+			}),
 		);
 
 		// Returns specific credentials
 		this.app.get(
 			`/${this.restEndpoint}/credentials/:id`,
-			ResponseHelper.send(
-				async (
-					req: express.Request,
-					res: express.Response,
-				): Promise<ICredentialsDecryptedResponse | ICredentialsResponse | undefined> => {
-					const { id } = req.params;
-					const queryBuilder = Db.collections.Credentials!.createQueryBuilder('c');
-					queryBuilder.andWhere('c.id = :id', { id });
-					queryBuilder.innerJoin('c.shared', 'shared');
-					queryBuilder.andWhere('shared.userId = :userId', {
-						userId: (req.user as User).id,
-					});
+			ResponseHelper.send(async (req: CredentialRequest.Get) => {
+				const { id: credentialId } = req.params;
 
-					// Make sure the variable has an expected value
-					const includeData = ['true', true].includes(req.query.includeData as string);
-
-					if (!includeData) {
-						queryBuilder.select([
-							'c.id',
-							'c.name',
-							'c.type',
-							'c.nodesAccess',
-							'c.createdAt',
-							'c.updatedAt',
-						]);
-					}
+				const shared = await Db.collections.SharedCredentials!.findOne({
```

Comment:
why not `findOneOrFail` ? then we don't need to do the check afterwards (line 1822)
```suggestion
				const shared = await Db.collections.SharedCredentials!.findOneOrFail({
```

Reply:
Changed. But this does not preserve the original failure response `{}`

Reply:
Reverting just in case.